Source/Core/Component/ComponentWithDelegate.swift

```diff
@@ -83,3 +83,34 @@ extension ComponentWithDelegate {
         componentDelegate.actions = actions
     }
 }
+
+extension ComponentWithDelegate {
+    /**
+     * Mutates the `componentState` only once using the closure provided.
+     * Should be used if you're modifying more than one of the `componentState`'s fields.
+     *
+     * **Usage**:
+     * ```
+     * var mutableState = componentState
+     * mutableState.name = newName
+     * mutableState.password = randomPassword
+     * mutableState.friends = [] as [Friend]
+     * componentState = mutableState
+     * ```
+     * becomes
+     * ```
+     * mutateState { state in
+     *   state.name = newName
+     *   state.password = randomPassword
+     *   state.friends = [allUsers] as [Friend]
```

Comment:
Slightly inconsistent with the previous usage, there was no `allUsers` array.

Reply:
Yeah, I wanted to signify that if you use `mutateState`, everyone will want to be your friend.